src/quo/components/tabs/tabs/view.cljs

```diff
@@ -72,31 +71,33 @@
            scroll-on-press?
            size
            style]}]
-  [rn/view
-   {:style (style/tab {:size             size
-                       :default-tab-size default-tab-size
-                       :number-of-items  number-of-items
-                       :index            index
-                       :style            style})}
-   [tab/view
-    {:id                  id
-     :notification-dot?   notification-dot?
-     :customization-color customization-color
-     :accessibility-label accessibility-label
-     :size                size
-     :blur?               blur?
-     :active              (= id @active-tab-id)
-     :on-press            (fn [id]
-                            (reset! active-tab-id id)
-                            (when (and scroll-on-press? @flat-list-ref)
-                              (.scrollToIndex ^js @flat-list-ref
-                                              #js
-                                               {:animated     true
-                                                :index        index
-                                                :viewPosition 0.5}))
-                            (when on-change
-                              (on-change id)))}
-    label]])
+  (let [on-press (rn/use-callback (fn [tab-id]
+                                    (set-active-tab-id tab-id)
+                                    (when (and scroll-on-press? flat-list-ref)
+                                      (.scrollToIndex ^js flat-list-ref
+                                                      #js
+                                                       {:animated     true
+                                                        :index        index
+                                                        :viewPosition 0.5}))
+                                    (when on-change
+                                      (on-change tab-id)))
+                                  [flat-list-ref])]
```

Comment:
or actually, seems it's better to move this function outside, then you can keep ref for flat-list-ref

Reply:
yeah, it would be best to just pass `tab-view` just a `on-press` prop instead of passing all of: `flat-list-ref`, `set-active-tab-id`, `scroll-on-press`, `on-change`.